warehouse/query-core/src/test/java/datawave/query/index/lookup/TupleToRangeTest.java

```diff
@@ -1,87 +1,235 @@
+        TupleToRange tupleToRange = new TupleToRange(queryNode, config);
+        Iterator<QueryPlan> ranges = tupleToRange.apply(tuple);
+        
+        // Assert ranges against expected ranges
+        eval(expectedRanges, ranges);
+    }
+    
+    @Test
+    public void testApplyWithTldDocumentRange() {
+        String shard = "20190314_0";
+        Set<String> docIds = Sets.newHashSet("docId0", "docId1", "docId2");
+        IndexInfo indexInfo = new IndexInfo(docIds);
+        indexInfo.applyNode(queryNode);
+        Tuple2<String,IndexInfo> tuple = new Tuple2<>(shard, indexInfo);
+        
+        // Build expected shard ranges
+        List<Range> expectedRanges = new ArrayList<>(3);
+        expectedRanges.add(makeTldTestRange(shard, "docId0"));
+        expectedRanges.add(makeTldTestRange(shard, "docId1"));
+        expectedRanges.add(makeTldTestRange(shard, "docId2"));
+        
+        // Create the ranges
+        config.setTldQuery(true);
+        TupleToRange tupleToRange = new TupleToRange(queryNode, config);
+        Iterator<QueryPlan> ranges = tupleToRange.apply(tuple);
+        
+        // Assert ranges against expected ranges
+        eval(expectedRanges, ranges);
+    }
+    
+    @Test
+    public void testApplyWithShardRange() {
+        String shard = "20190314_0";
+        IndexInfo indexInfo = new IndexInfo(-1);
+        indexInfo.applyNode(queryNode);
+        Tuple2<String,IndexInfo> tuple = new Tuple2<>(shard, indexInfo);
+        
+        // Build expected shard ranges
+        List<Range> expectedRanges = new ArrayList<>(1);
+        expectedRanges.add(makeShardedRange(shard));
+        
+        // Create the ranges
+        TupleToRange tupleToRange = new TupleToRange(queryNode, config);
+        Iterator<QueryPlan> ranges = tupleToRange.apply(tuple);
+        
+        // Assert ranges against expected ranges
+        eval(expectedRanges, ranges);
+    }
+    
+    @Test
+    public void testApplyWithDayRange() {
+        String shard = "20190314";
+        IndexInfo indexInfo = new IndexInfo(-1);
+        indexInfo.applyNode(queryNode);
+        Tuple2<String,IndexInfo> tuple = new Tuple2<>(shard, indexInfo);
+        
+        // Build expected shard ranges
+        List<Range> expectedRanges = new ArrayList<>(1);
+        expectedRanges.add(makeDayRange(shard));
+        
+        // Create the ranges
+        TupleToRange tupleToRange = new TupleToRange(queryNode, config);
+        Iterator<QueryPlan> ranges = tupleToRange.apply(tuple);
+        
+        // Assert ranges against expected ranges
+        eval(expectedRanges, ranges);
+    }
+    
+    private void eval(List<Range> expectedRanges, Iterator<QueryPlan> ranges) {
+        Iterator<Range> expectedIter = expectedRanges.iterator();
+        while (expectedIter.hasNext()) {
+            
+            Range expectedRange = expectedIter.next();
+            Range generatedRange = ranges.next().getRanges().iterator().next();
```

Comment:
Awesome tests! This is small so feel free to disregard. We are missing a scenario where the iterator returned from `QueryPlan.getRanges()` returns more than one result. We could consider ensuring that the iterator `hasNext()` returns false after we pull out the one range.

Reply:
I'll add both of those cases

Reply:
I get that. I'm saying the Iterator returned by `ranges.next().getRanges().iterator()` is never checked to ensure it has no more elements. If you captured it in a variable `Iterator<Range> generatedIter = ranges.next().getRanges().iterator()`, then you could do a final `assertFalse(generatedIter.hasNext())` if you wanted.

Reply:
Ah, I see what you're saying -- I'll make that assertion.